db/snapshot/backup_task.cc

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ *
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include "utils/lister.hh"
+#include "utils/s3/client.hh"
+#include "replica/database.hh"
+#include "db/config.hh"
+#include "db/snapshot-ctl.hh"
+#include "db/snapshot/backup_task.hh"
+#include "schema/schema_fwd.hh"
+#include "sstables/sstables.hh"
+
+extern logging::logger snap_log;
+
+namespace db::snapshot {
+
+backup_task_impl::backup_task_impl(tasks::task_manager::module_ptr module, snapshot_ctl& ctl, shared_ptr<s3::client> client, sstring bucket, sstring ks, sstring snapshot_name) noexcept
+    : tasks::task_manager::task::impl(module, tasks::task_id::create_random_id(), 0, "node", ks, "", "", tasks::task_id::create_null_id())
+    , _snap_ctl(ctl)
+    , _client(std::move(client))
+    , _bucket(std::move(bucket))
+    , _ks(std::move(ks))
+    , _snapshot_name(std::move(snapshot_name))
+{}
+
+std::string backup_task_impl::type() const {
+    return "backup";
+}
+
+tasks::is_internal backup_task_impl::is_internal() const noexcept {
+    return tasks::is_internal::no;
+}
+
+static std::pair<sstring, table_id> extract_cf_name_and_uuid(const sstring& directory_name) {
+    // cf directory is of the form: 'cf_name-uuid'
```

Comment:
probably would be better if the comment reads:

> // cf directory is of the form: <cf_name>-<uuid>

Reply:
I copy-n-paste-d it from replica/database.cc. Probably need to place it in some generic place as well

Reply:
we have `extract_cf_name_and_uuid` in replica/database.cc
wouldn't it better to share the code?  maybe under utils/

Reply:
Made a unified helper in v2